Add pypi classifiers and use .rst format for the Readme (#97)
eduNEXT/eox-core: README.rst

> =======================

Reviewer: I think we should add a Changelog or Release file to start tracking all the changes.
We should also start naming the releases or add a description on the GitHub release page. @morenol What do you think?

Author: Yes, I also thing that we need a Changelog

Author: We can add something like this to the cookiecutter, what do you thing?

https://github.com/edx/edx-cookiecutters/blame/master/python-template/%7B%7Bcookiecutter.placeholder_repo_name%7D%7D/CHANGELOG.rst

I can add something like that to this repository

Reviewer: Yes, I agree. @morenol

Author: Please, review that in https://github.com/eduNEXT/cookiecutter-openedx-plugin/pull/9 and then I will create other PR in this repo @Squirrel18

Reviewer: Ok.

EOX core |build-status|
=======================

.. |build-status| image:: https://circleci.com/gh/eduNEXT/eox-core.svg?style=svg

> Eox-core (A.K.A. Edunext Open eXtensions) is a `django app plugin](https://github.com/edx/edx-platform/tree/master/openedx/core/djangoapps/plugins) for adding multiple APIS for the [edx-platform <https://github.com/edx/edx-platform>`_ for adding multiple APIS for the `edx-platform <https://github.com/edx/edx-platform>`_ to make changing the platform easier on commercial environments, including bulk creation of pre-activated users (e.g. skip sending an activation email) and enrollments.

Reviewer: @morenol  This description is a bit weird, what do you think of:

Eox-core (A.K.A. Edunext Open extensions) is a plugin (plugins-link) for the edx-platform that adds multiple API endpoints in order to extend the functionality of the edx-platform and avoid changing the base code directly. These API endpoints includes bulk creation of pre-activated users (for example, skip sending an activation email) and enrollments.

Author: I just took that from the old README, I didn't add anything

Author: But I will add your changes

Usage
=====

> 1) Create the oauth client at http://localhost:18000/admin/oauth2/client/add/ , copy the client-id and client-secret.

Reviewer: Comma character is not correctly placed. @morenol

2) Generate an auth-token using that client-id and client-secret:

.. code-block:: bash

	$ curl -X POST -d "client_id=<YOUR_CLIENT_ID>&client_secret=<YOUR_CLIENT_SECRET>
		&grant_type=client_credentials" http://localhost:18000/oauth2/access_token/

3) Use the token to call the API as you need:

.. code-block:: bash

	# User creation API example
	curl -X POST --header "Authorization: Bearer <YOUR_AUTH_TOKEN>" -H "Accept: application/json" \
		http://localhost:18000/eox-core/api/v1/user/ --header "Content\-Type: application/json" \
		--data  '{"username": "jsmith", "email": "jhon@example.com", "password": "qwerty123",
		"fullname": "Jhon Smith"}'

	# Enroll api example
	curl -X POST --header "Authorization: Bearer <YOUR_AUTH_TOKEN>" -H "Accept: application/json" \
		http://localhost:18000/eox-core/api/v1/enrollment/ --header "Content-Type: application/json" \
		--data '{"course_id": "course-v1:edX+DemoX+Demo_Course", "email": "edx@example.com",
		"mode": "audit", "force": 1}'

Installation on Devstack
========================

> - Create an openedx Ironwood instace if it doesn't exist already:

Reviewer: Is this necessary? Why not just add the link to the devstack instructions? @morenol

	export OPENEDX_RELEASE=ironwood.master
	make dev.checkout
	make dev.clone
	make dev.provision
	echo export OPENEDX_RELEASE=ironwood.master > .envrc

- Clone the git repo:

.. code-block:: bash

	cd ~/Documents/eoxstack/src/  # Change path as needed
	sudo mkdir edxapp
	cd edxapp
	git clone git@github.com:eduNEXT/eox-core.git

- Install plugin from your server (in this case the devstack docker lms shell):

.. code-block:: bash

	cd ~/Documents/eoxstack/devstack  # Change for your devstack path (if you are using devstack)
	make lms-shell  # Enter the devstack machine (or server where lms process lives)
	sudo su edxapp -s /bin/bash  # if you are using devstack you need to use edxapp user
	source /edx/app/edxapp/venvs/edxapp/bin/activate
	cd /edx/src/edxapp/eox-core
	pip install -e .

Development
===========

To update dependencies you must edit the .in files (not requirements.txt directly) and compile them using pip-tools:

.. code-block:: bash

	$ source /path/to/venv/bin/activate
	(venv)$ pip install pip-tools
	(venv)$ pip-compile

Integrations with third party services
======================================

The plugin offers some integrations listed below:

> 1. **Sentry**: this service allows to track the errors generated on edx-platform. Check more details in https://sentry.io/welcome/. To enable the integration, follow the steps below:

Reviewer: Capitalize 'this'. @morenol

	* Install the plugin with Sentry support (extras_require [sentry])

	* Sign up/in to your sentry account and create a new Django application integration.

	* Get the DSN for your integration. This is an unique identifier for your application.

> 	* Setup the following configuration values for edx\-platform:

Reviewer: Remove the slash character. @morenol

		.. code-block:: yaml

			EOX_CORE_SENTRY_INTEGRATION_DSN: <your DSN value>
			EOX_CORE_SENTRY_IGNORED_ERRORS: [] # optional

	By default, **EOX_CORE_SENTRY_INTEGRATION_DSN** setting is None, which disables the sentry integration.

> 	**EOX_CORE_SENTRY_IGNORED_ERRORS** is optional. It is a list of exceptions that wants to be ignored by sentry. For instance, it can be defined as:

Reviewer: ```suggestion
	**EOX_CORE_SENTRY_IGNORED_ERRORS** is optional. It is a list of the exceptions you want to ignore. For instance, it can be defined as:
```

		.. code-block:: yaml

			EOX_CORE_SENTRY_IGNORED_ERRORS: [
				'xmodule.exceptions.NotFoundError',
				'openedx.core.djangoapps.user_authn.exceptions.AuthFailedError',
			]

> Course Management automation compilation

Reviewer: Period at the end of the line. @morenol

Reviewer: @morenol  We can add a description like:

This component allows Studio users to make changes in multiple courses, such as:
- Add or remove staff/instructor users across multiple courses in one organization.
- Change course settings in multiple courses at once.
- Re-run a course across multiple organizations.

Author: I think that the period is not needed here since this is a title.

On the other hand, is that description related to `Course Management automation compilation`? @Squirrel18

Reviewer: Let's rename this section to: Course Management automation and then create a new subsection containing the compilation instructions. @morenol

========================================

We use webpack to bundle the React js application and its dependencies,

> 

Reviewer: Remove this blank space and add a period instead of the comma. @morenol

> in order to compile for dev environment run this command in the root folder:

Reviewer: ```suggestion
To compile in a development environment, run this command on the root folder:
```

npm run build-dev

> Otherwise, if you want to compile for use in production environment, run this command instead:

Reviewer: ```suggestion
Otherwise, if you want to compile it for use in a production environment, run this command instead:
```

Reviewer: What about this? @morenol

npm run build-prod

> This command is defined in the package.json file and export two bundles (a build.js and course-management.bundle.css) inside of eox_core/static folder.

Reviewer: ```suggestion
These commands are defined in the package.json file and each one exports two bundle files (build.js and course-management.bundle.css) inside of eox_core/static folder.
```

EOX core migration notes
========================

**Migrating to version 2.0.0**

From version **2.0.0**, middlewares **RedirectionsMiddleware** and **PathRedirectionMiddleware** are now included in this plugin. These middlewares were moved from the **eox-tenant** plugin `here <https://github.com/eduNEXT/eox-tenant/>`_.

if you installed **eox-core** alongside **eox-tenant** plugin, follow the steps below:

- Upgrade eox-tenant to version **1.0.0** (previous releases are not compatible with eox-core 2.0.0)
- Run the plugin migrations as indicated below:

.. code-block:: bash

   $ python manage.py lms migrate eox_tenant --settings=<your app settings>
   $ python manage.py lms migrate eox_core --fake-initial --settings=<your app settings>

> In the case eox-tenant is not installed in the platform, just run the eox-core migrations normally.

Reviewer: ```suggestion
In case eox-tenant is not installed on the platform, just run the eox-core migrations.
```
